src/server/index_patterns/service/lib/field_capabilities/field_caps_response.js

```diff
@@ -63,27 +64,45 @@ export function readFieldCapsResponse(fieldCapsResponse) {
     const capsByType = capsByNameThenType[fieldName];
     const types = Object.keys(capsByType);
 
+    // If there are multiple types but they all resolve to the same kibana type
+    // ignore the conflict and carry on (my wayward son)
     if (types.length > 1) {
```

Comment:
No reason to check the length of types here, just get the `uniqueKibanaTypes` and check that.